app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncUser.java

```diff
@@ -207,4 +215,54 @@ public void done(@Nullable JSONObject jsonObject) {
         new NetworkRequest(template, callback).execute();
         return promise;
     }
+
+    public static Promise uploadProfilePhoto(String photoName, byte[] data) {
+
+        final Promise promise = new Promise();
+
+        NetworkRequest.Callback callback = new NetworkRequest.Callback() {
+            @Override
+            public void done(@Nullable JSONObject jsonObject) {
+                if (jsonObject == null) {
+                    Log.e(TAG, "Error getting uploading photo response json");
+                    promise.onFailure();
+                    return;
+                }
+                Log.d(TAG, "uploading image response:" + jsonObject.toString());
+                try {
+                    String photoName = jsonObject.getString("name");
+                    // parse photo json, get photo "name" and update user photo
+                    // TODO: refactor using bolts
+                    SharedPreferences sharedPreferences =
+                            MyApplication.getContext().getSharedPreferences(
+                                    MyApplication.getContext()
+                                            .getString(R.string.login_or_sign_up_session), 0);
+                    String userId = sharedPreferences.getString(SyncUser.JsonKeys.USER_ID, null);
+                    NetworkRequest.Callback callbackUpdateProfilePhoto = new NetworkRequest.Callback() {
+                        @Override
+                        public void done(@Nullable JSONObject jsonObject) {
+                            if (jsonObject == null) {
+                                Log.e(TAG, "Error getting updating photo response json");
+                                promise.onFailure();
+                                return;
+                            }
+                            Log.d(TAG, "updating image response:" + jsonObject.toString());
+                        }
+                    };
+
+                    UrlTemplate templateUpdateProfilePhoto = UrlTemplateCreator.updateProfilePhoto(userId, photoName);
+                    new NetworkRequest(templateUpdateProfilePhoto, callbackUpdateProfilePhoto).execute();
+                    promise.onSuccess();
```

Comment:
I think you should wait until the profile update callback is finished to call promise.onSuccess() because the user is in interested in making sure their profile picture is set, not that it was uploaded to the server.